packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -1,35 +1,43 @@
-import { createElement, ReactElement, useReducer, useRef } from "react";
+import { createElement, Dispatch, ReactElement, useCallback, useReducer, useRef } from "react";
 
-import AccordionGroup, { AccGroup } from "./AccordionGroup";
-import { getAccordionGroupsReducer } from "../utils/AccordionGroupStateReducer";
+import AccordionGroup, { AccordionGroupProps } from "./AccordionGroup";
+import {
+    CollapsedAccordionGroupsReducerAction,
+    getCollapsedAccordionGroupsReducer
+} from "../utils/CollapsedAccordionGroupsReducer";
 import { AccordionContainerProps } from "../../typings/AccordionProps";
 import classNames from "classnames";
 
+export type AccordionGroups = Array<Pick<AccordionGroupProps, "header" | "content" | "visible" | "dynamicClassName">>;
+
 export interface AccordionProps extends Pick<AccordionContainerProps, "class" | "style" | "tabIndex"> {
     id: string;
-    groups: AccGroup[];
+    groups: AccordionGroups;
     collapsible: boolean;
+    animateCollapsing?: boolean;
     singleExpandedGroup?: boolean;
+    generateHeaderIcon?: (collapsed: boolean) => ReactElement;
+    showGroupHeaderIcon?: "right" | "left" | "no";
 }
 
 export default function Accordion(props: AccordionProps): ReactElement | null {
-    const previousGroupsPropValue = useRef(props.groups);
+    const reducer = useRef(getCollapsedAccordionGroupsReducer(props.singleExpandedGroup ? "single" : "multiple")); // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
 
-    const [accordionGroups, accordionGroupsDispatch] = useReducer(
-        getAccordionGroupsReducer(props.collapsible ? (props.singleExpandedGroup ? "single" : "multiple") : "all"), // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
-        props.groups.map(group => ({ ...group, collapsed: props.collapsible }))
+    const [collapsedAccordionGroups, collapsedAccordionGroupsDispatch] = useReducer(
```

Comment:
NOTE: I have the feeling the reducer is not needed at all based on our chat this morning about the data change while using conditional visible tabs.

Reply:
I think the reducer is still useful because it hides the complexity of updating state logic with regard to the different expand modes. It's also a nice abstraction to be shared by both web and native widget in the end.